### PR TITLE
Jetpack Connect: Introduce a Site Topic step

### PR DESCRIPTION
**Pull request by tyxla**
This PR introduces a site topic step, based on the corresponding WP.com step UI.

This doesn't hook up this new step in the flows, it only makes the step available. We'll be hooking the steps together in #30905.

#### Changes proposed in this Pull Request

* Introduce a site topic step

#### Testing instructions

* Spin up calypso.live off this branch.
* Go to `/jetpack/site-topic/:site` where `:site` is the slug of a connected Jetpack site.
* Input/select a site topic.
* Click the button to save it.
* Go to `/wp-admin/options.php` of your site and witness a `site_vertical` option with the selected value.

### CODE REVIEW

File: client/jetpack-connect/site-topic.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		// @TODO: implement saving logic
+	}
+
+	render() {
+		const { translate } = this.props;
+
+		return (
+			<MainWrapper isWide>
+				<div className="jetpack-connect__step">
+					<FormattedHeader headerText={ translate( 'Tell us about your website?' ) } />
```
Comment (a8ci18n) [inline client/jetpack-connect/site-topic.js]: Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Tell us about your website' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (tyxla) [reply client/jetpack-connect/site-topic.js]: Thanks @a8ci18n, great suggestion! Fixed in https://github.com/Automattic/wp-calypso/pull/30731/commits/5dc155f23b41bf3c057a234961494d6dc0439134

**Comment by keoshi** [comment]
This is the only step that doesn't have a description. Should we introduce one and say something along these lines “What topic or category does your site cover?” ?

cc @joanrho @jeffgolenski

**Comment by tyxla** [comment]
I think this could use a description indeed 👍 There's more than enough space for that. Let me know what copy you prefer and I'll add it accordingly.

**Comment by joanrho** [comment]
@keoshi I like your suggestion for the description! It helps clarify what users should think about putting in the text field. Let's ping @Automattic/editorial for review of that in case they might have other suggestions.

**Comment by AnnaMag** [comment]
Code looks good, but data saving does not work me (checked in the `/options.php` and via CLI). 
@tyxla could you please test?

**Comment by AnnaMag** [comment]
Re my earlier comment, just realized we need `'manage_options'` permission for the data saving to work.

**Comment by tyxla** [comment]
> just realized we need `'manage_options'` permission for the data saving to work.

@AnnaMag the endpoint does this check already. And yes, we need this capability, otherwise the user will be very limited in what they can do anyway 😉

**Comment by tyxla** [comment]
> @tyxla could you please test?

Just verified it works well btw 👍 Seems like the only reason on your side was missing the `manage_options` capability.

**Comment by AnnaMag** [comment]
yes, I recall implementing it :) The note was re me testing it on a site without permissions, thus failure saving the data.

Re-rested with an admin role, works well 👍

**Comment by tyxla** [comment]
Just added the description @keoshi recommended:



@keoshi what do you think? Can I get design ✅ for this?

I say let's ship this, and we can later iterate if @Automattic/editorial folks have a suggestion for improving this.

**Comment by keoshi** [comment]
@tyxla Would it be ok to wait for a couple of hours while I tag this as “needs copy review” and we wait a little bit for someone from Editorial?

re: Design: I'll take another look right away!

**Comment by tyxla** [comment]
> Would it be ok to wait for a couple of hours while I tag this as “needs copy review” and we wait a little bit for someone from Editorial?

Sure thing, I don't mind waiting 👍 

> re: Design: I'll take another look right away!

Thanks!

**Comment by michelleweber** [comment]
I'm thinking "What is your site about?"/"What will your site be about?" or "Tell us what your site is/will be about." I like @keoshi 's suggestions, but would like to so a bit less formal-sounding, if that works for y'all.

**Comment by tyxla** [comment]
Thanks @michelleweber - I'm fine with all these - I'll let @keoshi pick his favorite 😉

**Comment by keoshi** [comment]
@michelleweber Thanks for jumping on this so quickly, really appreciate it!

I think we would like to avoid any future tense because, since this is a Jetpack site, their website might already exist. From that perspective, “Tell us what your site is about” is great.

Also, for the description, I agree that my proposal — “What topic or category does your site cover” — sounds a bit formal, do you have any other suggestions?

Thanks again!

**Comment by keoshi** [comment]
I also see the ”Continue” button flickering as we type into the input, but I see that's how it works on dotcom as well, so it's unrelated to this PR.

**Comment by tyxla** [comment]
As discussed via Slack, we're removing the description for now, as it's a bit repetitive and redundant in the context where a similar title exists. I've opened #30982 where we can discuss what description to use - I'll add it as soon as we decide on one.

> Could we personalize this to match the JP brand like we've done for the site type step and modify the styles of the input?

Yes, great call! Done in https://github.com/Automattic/wp-calypso/pull/30731/commits/d050a1c25035880119d24e7e41c2a264756c2084

> I also see the ”Continue” button flickering as we type into the input, but I see that's how it works on dotcom as well, so it's unrelated to this PR.

Yeah, and I'd say that's also expected - we're querying the huge list of verticals for matches of the current input, and while loading, the button is disabled.

**Comment by tyxla** [comment]
Thanks everybody, I'll 🚢 when the tests are good.

**Comment by michelleweber** [comment]
Makes sense re: future tense. So something like: "Tell us a bit about your site -- what's it about? Do you have a particular focus? Enter a few keywords."

**Comment by keoshi** [comment]
Nice, I love that last bit: “Do you have a particular focus? Enter a few keywords.”

Adding that to the new issue Marin created to track this https://github.com/Automattic/wp-calypso/issues/30982

Thanks again, @michelleweber !

**Pull request merged**